src/app/components/tree/tree.component.html

```diff
@@ -1,11 +1,22 @@
 <ng-template #reverseItems let-items>
   <ul>
     <li *ngFor="let item of items" class="item">
-      <div>
-        <div class="item__name">{{item.name}}</div>
+      <div class="item__name"
+           [search]="inputValue"
+           [itemId]="item.id"
+           [attr.data-value]="item.id"
+           #search='search'
+      >
+        <input
+          type="checkbox"
+          [checked]="item.isSelected"
+          (change)="select(item)"
+          [disabled]="search.checkSelected()"
```

Comment:
don't use methods for inputs (it's not good practice). Declaring variable for isSelected and sending it to disable is better way